libraries/chain/son_evaluator.cpp

```diff
@@ -139,7 +139,7 @@ object_id_type son_heartbeat_evaluator::do_apply(const son_heartbeat_operation&
             is_son_active = false;
         }
 
-        if(itr->status == son_status::in_maintenance) {
+        if((itr->status == son_status::in_maintenance) || (itr->status == son_status::request_maintenance)) {
```

Comment:
request_maintenance son can also send heartbeats till the maintenance, with this check I think we will move it to active or inactive and loose the state.

Reply:
I think it is ok to return to the active mode from request_maintenance state with heartbeat message. Our goal is to minimize the time when the SON can't verify payments,  so if it need a maintenance time and complete the maintenance work before the maintenance interval it should have opportunity to return to active state without waiting for maintenance block.

Reply:
Actually it is expected that the SON continues to send heartbeats in request_maintenance state. So it is expected that the SON continues to work in full capacity till the next chain maintenance.